1.2.0: test suite fails in `tests/compat/test_responses_load_cache.py::test_mock_session` unit with pytest 8.2.1

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix> using `installer` module
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

<details>
<summary>Here is pytest output:</summary>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-requests-cache-1.2.0-2.fc37.x86_64/usr/lib64/python3.10/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-requests-cache-1.2.0-2.fc37.x86_64/usr/lib/python3.10/site-packages
+ /usr/bin/pytest -ra -m 'not network' --ignore tests/integration
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.10.14, pytest-8.2.1, pluggy-1.5.0
rootdir: /home/tkloczko/rpmbuild/BUILD/requests-cache-1.2.0
configfile: pyproject.toml
plugins: time-machine-2.14.1, requests-mock-1.12.1, rerunfailures-12.0
collected 381 items

tests/compat/test_requests_mock_combine_cache.py .                                                                                                                                    [  0%]
tests/compat/test_requests_mock_disable_cache.py .                                                                                                                                    [  0%]
tests/compat/test_requests_mock_load_cache.py .                                                                                                                                       [  0%]
tests/compat/test_responses_load_cache.py F                                                                                                                                           [  1%]
tests/unit/models/test_base.py ..                                                                                                                                                     [  1%]
tests/unit/models/test_raw_response.py ......                                                                                                                                         [  3%]
tests/unit/models/test_request.py .                                                                                                                                                   [  3%]
tests/unit/models/test_response.py .................                                                                                                                                  [  7%]
tests/unit/policy/test_actions.py ..........................................................................................                                                          [ 31%]
tests/unit/policy/test_expiration.py .......................                                                                                                                          [ 37%]
tests/unit/test_base_cache.py ......................                                                                                                                                  [ 43%]
tests/unit/test_cache_keys.py .............................                                                                                                                           [ 50%]
tests/unit/test_patcher.py ...........                                                                                                                                                [ 53%]
tests/unit/test_serializers.py .ss.s...                                                                                                                                               [ 55%]
tests/unit/test_session.py .......................................................................................................................................................... [ 96%]
..............                                                                                                                                                                        [100%]

========================================================================================= FAILURES ==========================================================================================
_____________________________________________________________________________________ test_mock_session _____________________________________________________________________________________

mock_http_adapter = <MagicMock name='get_connection' id='140212715324224'>

    @patch.object(
        requests.adapters.HTTPAdapter, 'get_connection', side_effect=ValueError('Real request made!')
    )
    def test_mock_session(mock_http_adapter):
        """Test that the mock_session fixture is working as expected"""
        with get_responses():
            # An error will be raised if a real request is made
>           with pytest.raises(ValueError):
E           Failed: DID NOT RAISE <class 'ValueError'>

tests/compat/test_responses_load_cache.py:52: Failed
----------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------
[05-27 12:57:14] INFO     request.allowed-passthru              __init__.py:1078
------------------------------------------------------------------------------------- Captured log call -------------------------------------------------------------------------------------
INFO     responses:__init__.py:1078 request.allowed-passthru
================================================================================== short test summary info ==================================================================================
SKIPPED [1] tests/unit/test_serializers.py:39: ujson is not installed
SKIPPED [1] tests/unit/test_serializers.py:48: bson is not installed
SKIPPED [1] tests/unit/test_serializers.py:88: itsdangerous is not installed
FAILED tests/compat/test_responses_load_cache.py::test_mock_session - Failed: DID NOT RAISE <class 'ValueError'>
========================================================================= 1 failed, 377 passed, 3 skipped in 7.26s ==========================================================================
```
</details>

<details>
<summary>List of installed modules in build env:</summary>

```console
Package                       Version
----------------------------- -----------
alabaster                     0.7.16
async-timeout                 4.0.3
attrs                         23.2.0
Babel                         2.15.0
boto3                         1.34.113
botocore                      1.34.113
build                         1.2.1
cattrs                        23.2.3
charset-normalizer            3.3.2
contourpy                     1.2.1
cycler                        0.12.1
defusedxml                    0.7.1
docutils                      0.20.1
exceptiongroup                1.1.3
fastjsonschema                2.19.1
fonttools                     4.51.0
idna                          3.7
imagesize                     1.4.1
importlib_metadata            7.1.0
iniconfig                     2.0.0
installer                     0.7.0
Jinja2                        3.1.4
jmespath                      1.0.1
kiwisolver                    1.4.5
lark                          1.1.9
linkify-it-py                 2.0.3
markdown-it-py                3.0.0
MarkupSafe                    2.1.5
matplotlib                    3.8.4
mdit-py-plugins               0.4.1
mdurl                         0.1.2
myst-parser                   3.0.1
numpy                         1.26.4
olefile                       0.47
packaging                     24.0
pbr                           5.11.1
pillow                        10.3.0
platformdirs                  4.2.2
pluggy                        1.5.0
poetry-core                   1.9.0
psutil                        5.9.8
Pygments                      2.18.0
pyparsing                     3.1.2
pyproject_hooks               1.0.0
pytest                        8.2.1
pytest-rerunfailures          12.0
python-dateutil               2.9.0.post0
PyYAML                        6.0.1
redis                         5.0.4
requests                      2.32.2
requests-mock                 1.12.1
responses                     0.25.0
rich                          13.7.1
s3transfer                    0.10.1
snowballstemmer               2.2.0
Sphinx                        7.3.7
sphinx-autodoc-typehints      2.1.0
sphinx-automodapi             0.17.0
sphinx-copybutton             0.5.2
sphinx_design                 0.6.0
sphinx-notfound-page          1.0.2
sphinxcontrib-apidoc          0.5.0
sphinxcontrib-applehelp       1.0.8
sphinxcontrib-devhelp         1.0.6
sphinxcontrib-htmlhelp        2.0.5
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.7
sphinxcontrib-serializinghtml 1.1.10
sphinxext-opengraph           0.9.1
tenacity                      8.3.0
time-machine                  2.14.1
timeout-decorator             0.5.0
tokenize_rt                   5.2.0
tomli                         2.0.1
typing_extensions             4.12.0
uc-micro-py                   1.0.3
url-normalize                 1.4.3
urllib3                       2.2.1
wheel                         0.43.0
zipp                          3.19.0
```
</details>

Please let me know if you need more details or want me to perform some diagnostics.

--- a/HISTORY.md
+++ b/HISTORY.md
@@ -6,6 +6,7 @@
 * Fix `normalize_headers` not accepting header values in bytes
 * Fix inconsistency due to rounding in `CachedResponse.expires_unix` property
 * Fix form boundary used for cached multipart requests to _fully_ comply with RFC 2046
+* Fix usage example with `responses` library to be compatible with `requests` 2.32
 
 ## 1.2.0 (2024-02-17)
 

--- a/tests/compat/test_responses_load_cache.py
+++ b/tests/compat/test_responses_load_cache.py
@@ -44,7 +44,9 @@ def get_responses():
 
 # responses patches HTTPAdapter.send(), so we need to patch one level lower to verify request mocking
 @patch.object(
-    requests.adapters.HTTPAdapter, 'get_connection', side_effect=ValueError('Real request made!')
+    requests.adapters.HTTPAdapter,
+    'get_connection_with_tls_context',
+    side_effect=ValueError('Real request made!'),
 )
 def test_mock_session(mock_http_adapter):
     """Test that the mock_session fixture is working as expected"""

